### PR TITLE
[1542] Add edit provider feature (admin only)

### PR DESCRIPTION
**Pull request by avinhurry**
### Context

Adding the ability to edit providers after they are created in System Admin.

### Changes proposed in this pull request

We are adding a link to the provider show page, which takes you to a form where you can edit the providers details.

- Add edit and update controller actions on provider
- Add link to provider view
- Add edit provider view page
- Add translation for edit page title
- Add `/edit` route for provider
- Erb linter has made some fixes

### Guidance to review

- Sign in to Register as an Admin, i.e `Adam Baker`  https://rtt-review-pr-867.herokuapp.com

- Click `System admin` at the top right of the page, within the navigation bar

- Click on one of the providers and click the new `Edit this provider` link

- Change something, e.g `Provider name` and click continue

- Check that the name has been updated in your list of providers

### Screenshots

image

image

### CODE REVIEW

**Comment by avinhurry** [comment]
[[S] Edit provider](https://trello.com/c/2tFmR66A/1542-s-edit-provider)

File: app/controllers/system_admin/providers_controller.rb [hunk 1]
```diff
@@ -25,6 +25,19 @@ def show
       @users_view = UsersView.new(@provider)
     end
 
+    def edit
```
Comment (mooktakim) [inline app/controllers/system_admin/providers_controller.rb:28]: Its better to have a before_action here to find the provider. This should be for show, edit, update actions.

File: app/views/system_admin/providers/show.html.erb [hunk 2]
```diff
@@ -9,6 +9,8 @@
 
 <h1 class="govuk-heading-l"><%= @provider.name %></h1>
 
+<p> <%= govuk_link_to "Edit this provider", edit_provider_path, { class: "govuk-link govuk-link--no-visited-state" } %> </p>
```
Comment (mooktakim) [inline app/views/system_admin/providers/show.html.erb:12]: I think this is better as a secondary button next to the "Add a user" button.

File: config/routes/system_admin_routes.rb [hunk 3]
```diff
@@ -4,7 +4,7 @@ module SystemAdminRoutes
   def self.extended(router)
     router.instance_exec do
       scope module: :system_admin, path: "system-admin" do
-        resources :providers, except: %i[edit update destroy] do
+        resources :providers, except: %i[destroy] do
```
Comment (mooktakim) [inline config/routes/system_admin_routes.rb:7]: Can you change this to `only` instead of `except`. Its better to be explicit about which actions we want.

**Review by mooktakim — CHANGES_REQUESTED** [review]

File: app/controllers/system_admin/providers_controller.rb (hunk 1 above)
Comment (avinhurry) [reply app/controllers/system_admin/providers_controller.rb:28]: sorted

**Review by mooktakim — APPROVED** [review]

**Comment by GeoffJohnson** [comment]
All good

**Pull request merged**